app/src/main/java/com/mapbox/maps/testapp/examples/localization/LocalizationSettingsActivity.kt

```diff
@@ -0,0 +1,308 @@
+ * and set [MapboxCommonSettings.LANGAUGE] or [MapboxCommonSettings.WORLDVIEW].
+ *
+ * [SettingsServiceInterface] stores the provided language/worldview as key:value pair in DataStore
+ * and try to apply it to the tiles.if the values are supported,
+ * the localized tiles will be reloaded on to the map,
+ * else server will emit 400 error which can be listen using [OnMapLoadErrorListener] event.
+ *
+ * The language param should conform to BCP-47 tag and Worldview param to ISO- 3166-1 alpha-2 strings.
+ *
+ * This is currently an opt-in and experimental feature, i.e. some styles could have a minor visual changes.
+ *
+ * Previous localization functionality using [StyleInterface.localizeLabel] will not work with the new settings api.
+ * it will be deprecated in the future in order to benefit from more languages and additional worldview support.
+ * In case If you don't want to use server side localization and its already set, it can be removed
+ * using [SettingsServiceInterface.erase] function.
+ */
+class LocalizationSettingsActivity : AppCompatActivity(), OnMapLoadErrorListener, CoroutineScope {
+  private lateinit var mapboxMap: MapboxMap
+  private lateinit var languageSpinner: Spinner
+  private lateinit var worldViewSpinner: Spinner
+  private val supportedLanguageList = mutableListOf<LocalizationData>()
+  private val supportedWorldviewList = mutableListOf<LocalizationData>()
+  private val job = Job()
+
+  /**
+   * In order to update localization settings, Initialize settings interface as a Persistent.
+   * the language and worldview changes will persist across application lifecycle.
+   */
+  private val settingsServiceInterface: SettingsServiceInterface by lazy {
+    SettingsServiceFactory.getInstance(
+      SettingsServiceStorageType.PERSISTENT
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityLocalizationSettingBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+
+    languageSpinner = binding.languageView
+    worldViewSpinner = binding.worldView
+    mapboxMap = binding.mapView.getMapboxMap()
+    init()
+
+    // check applied settings
+    binding.buttonCheck.setOnClickListener {
+      getLanguageAndWorldViewSettings()
+    }
+  }
+
+  private fun init() {
+    mapboxMap.addOnMapLoadErrorListener(this)
+    mapboxMap.loadStyleUri(Style.MAPBOX_STREETS) { style ->
+      val source = vectorSource(SOURCE_ID) {
+        url("mapbox://mapbox.mapbox-terrain-v2")
+      }
+      style.addSource(source)
+    }
+    createLocalizationData()
+    getLanguageAndWorldViewSettings()
+    initSpinners()
+  }
+
+  /**
+   * Set language by providing the string value for language in BCP-47 format.
+   * Settings set/get are synchronous, make sure to run it on background thread.
+   */
+  private fun setLanguage(language: String): Expected<String, None> {
+    return settingsServiceInterface.set(LANGUAGE, Value(language))
```

Comment:
that you program against an interface is :+1: but that doesn't need to be included in naming parameters.
No added value in calling it `settingsServiceInterface` vs `settingsService`.